File: src/main/java/sc02/Q03.java
package sc02;

import java.io.BufferedReader;
import java.io.InputStreamReader;

/*
it is time to study -> study
Hello my name is Yaro -> Hello
 */

public class Q03 {
    public static void main(String[] args) {
        try (BufferedReader br = new BufferedReader(new InputStreamReader(System.in))) {
            String str = br.readLine();

            String answer = "";
            int pos = 0;
            int max = 0;
            while ((pos = str.indexOf(" ")) != -1) {
                String tmp = str.substring(0, pos);
                int len = tmp.length();
                if (len > max) {
                    max = len;
                    answer = tmp;
                }
                str = str.substring(pos + 1);
            }
            if (str.length() > max) {
                answer = str;
            }

            System.out.println(answer);
        } catch (Exception ex) {
            System.out.println("ex = " + ex);
        }

    }
}

File: src/main/kotlin/sc02/Q03.kt
package jp.co.kopher.sc02

import java.io.BufferedReader
import java.io.InputStreamReader

/*
it is time to study -> study
Hello my name is Yaro -> Hello
 */

class Q03

fun main() {
    try {
        BufferedReader(InputStreamReader(System.`in`)).use { br ->
            var str = br.readLine()

            var answer = ""
            var pos = 0
            var max = 0
            while ((str.indexOf(" ").also { pos = it }) != -1) {
                val tmp = str.substring(0, pos)
                val len = tmp.length
                if (len > max) {
                    max = len
                    answer = tmp
                }
                str = str.substring(pos + 1)
            }
            if (str.length > max) {
                answer = str
            }
            println(answer)
        }
    } catch (ex: Exception) {
        ex.printStackTrace()
    }
}